cmake/HPX_SetupStdexec.cmake

```diff
@@ -0,0 +1,34 @@
+if(STDEXEC_ROOT AND NOT Stdexec_ROOT)
+    set(Stdexec_ROOT
+        ${STDEXEC_ROOT}
+        CACHE PATH "STDEXEC base directory"
+    )
+    unset(STDEXEC_ROOT CACHE)
+endif()
+
+if(HPX_WITH_FETCH_STDEXEC AND NOT Stdexec_ROOT)
+    hpx_info(
+        "HPX_WITH_FETCH_STDEXEC=${HPX_WITH_FETCH_STDEXEC}, Stdexec will be fetched using CMake's FetchContent."
+    )
+    if(UNIX)
+        include(FetchContent)
+        FetchContent_Declare(
+            Stdexec
+            GIT_REPOSITORY https://github.com/NVIDIA/stdexec.git
+            GIT_TAG        main
```

Comment:
Can we make the fetched tag configurable, please?

Reply:
Of course!